src/object.d

```diff
@@ -3163,7 +3163,18 @@ inout(V) get(K, V)(inout(V[K])* aa, K key, lazy inout(V) defaultValue)
 ref V require(K, V)(ref V[K] aa, K key, lazy V value = V.init)
 {
     bool found;
-    auto p = cast(V*) _aaGetX(cast(void**)&aa, typeid(V[K]), V.sizeof, &key, found);
+    // if key and value are @safe-ly copyable, `require` can infer @safe
+    static if (isSafeCopyable!K && isSafeCopyable!V)
```

Comment:
How does this condition help, why not always use the `@trusted` wrapper?

Reply:
Maybe that wasn't that clear of an answer: if we just wrap the whole call into a @trusted, it's going to allow un-@safe key postblits. The unittest where S is key will fail.

Reply:
I still don't understand. Your running exactly the same code regardless if it's `isSafeCopyable` or not.

Reply:
Yes, the code is the same, but the inferred attributes for update()/require() will be different. _aaGetX internally calls the key's postblit. That postblit may be user-defined. If that postblit is `@system`, the whole require()/update() call should be `@system`. But if I were to wrap the call to _aaGetX into a `@trusted` unconditionally, I'd in effect silently promote `@system` user code to `@safe`.

Reply:
Ah, ok, I see. Thanks for explaining. Wasn't every obvious to me at least.

Reply:
No problem.

Reply:
This good to go?

Reply:
By me? Yes :)

Reply:
Please squash.